request_test.go

```diff
@@ -529,6 +529,32 @@ func TestBooleanRequestWithContextAndTimeout(t *testing.T) {
 	<-done
 }
 
+func TestWrongBodyResponse(t *testing.T) {
+	ts := newServer(response{200, "text/html", `
+		<html>
+		<header><title>This is title</title></header>
+		<body>
+		Hello world
+		</body>
+		</html>		
+	`})
+	defer ts.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond)
+	defer cancel()
+
+	cs := NewClient(ts.URL, "TOKEN", "SECRET")
+
+	_, err := cs.RequestWithContext(ctx, &ListZones{})
```

Comment:
you may drop the `ctx` as that's not the tested thing here.

Reply:
yep